src/audio/component.c

```diff
@@ -71,7 +71,7 @@ int comp_set_state(struct comp_dev *dev, int cmd)
 	if (dev->state == requested_state) {
 		comp_info(dev, "comp_set_state(), state already set to %u",
 			  dev->state);
-		return COMP_STATUS_STATE_ALREADY_SET;
+		return 0;
```

Comment:
@fkwasowi I think this might end up breaking ipc3. Why not check for the already _set return value in pipeline_task_cmd() for ipc4 and return 0 there?

Reply:
or alternatively check for the IPC version in comp_set_state() and return the appropriate return value for each IPC version ie COMP_STATUS_STATE_ALREADY_SET for IPC3 and 0 for IPC4